src/main/java/gov/cabinetoffice/gap/spotlight/publisher/service/RestService.java

```diff
@@ -0,0 +1,122 @@
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+public class RestService {
+
+    public static final String BACKEND_API_URL = System.getenv("BACKEND_API_URL");
+    public static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+
+    public static final Gson gson = new GsonBuilder().registerTypeAdapter(Instant.class,
+                    (JsonDeserializer<Instant>) (json, type, jsonDeserializationContext) -> OffsetDateTime
+                            .parse(json.getAsJsonPrimitive().getAsString()).toInstant())
+            .create();
+
+    private static final Logger logger = LoggerFactory.getLogger(RestService.class);
+
+    private static final String LAMBDA_AUTHORIZATION_SECRET = System.getenv("LAMBDA_AUTHORIZATION_SECRET");
+
+    public static <T> T sendGetRequest(OkHttpClient restClient, Map<String, String> params, String endpoint, Class<T> clazz) throws Exception {
+
+        HttpUrl.Builder httpBuilder = HttpUrl.get(BACKEND_API_URL + endpoint).newBuilder();
+        if (params != null) {
+            for (Map.Entry<String, String> param : params.entrySet()) {
+                httpBuilder.addQueryParameter(param.getKey(), param.getValue());
+            }
+        }
+
+        final Request request = defaultRequestBuilder().url(httpBuilder.build()).build();
+
+        try (Response response = restClient.newCall(request).execute()) {
+            if (response.isSuccessful()) {
+                logger.info("Successfully fetched from {}", endpoint);
+                return gson.fromJson(response.body().string(), clazz);
+            } else {
+                logger.info("Error occured while getting  {} with error {}, and body {}", endpoint, response.code(), response.body());
+                throw new RuntimeException();
+            }
+        }
+    }
+
+    public static <T> T sendPostRequest(OkHttpClient restClient, T requestBodyDto, String endpoint, Class<T> clazz) throws Exception {
+        RequestBody requestBody;
+        if (requestBodyDto != null) {
+            requestBody = RequestBody.create(gson.toJson(requestBodyDto), JSON);
+        } else {
+            requestBody = RequestBody.create("", JSON);
+        }
+        logger.info("Sending post request to {}", endpoint);
+
+        return executePost(restClient, requestBody, endpoint, clazz);
+    }
+
+    public static <T> T executePost(OkHttpClient restClient, RequestBody body, String endpoint, Class<T> clazz) throws Exception {
+        final Request request = defaultRequestBuilder().url(BACKEND_API_URL + endpoint).post(body).build();
+
+        try (Response response = restClient.newCall(request).execute()) {
+            if (response.isSuccessful()) {
+                logger.info("Successfully posted to {}", endpoint);
+
+                // you can only make one call to body.string before the stream is closed so don't refactor this out...
+                final String bodyString = response.body().string();
```

Comment:
response.body().string(); will throw a null pointer therefore 

if (bodyString == null) {
                    return null;
                }

will never execute in this case.  Consider refactoring.

Reply:
The JavaDoc for the `response.body` method says it should always be non-null provided the request was made through the `execute` method